lib/github_service/commands/run_test.rb

```diff
@@ -0,0 +1,302 @@
+        clone_repo_if_necessary
+        create_cross_repo_test_branch
+        update_travis_yaml_content
+        commit_travis_yaml_changes
+        push_commit_to_remote
+        create_cross_repo_test_pull_request
+      end
+
+      private
+
+      def _execute(issuer:, value:)
+        @issuer = issuer
+        parse_value(value)
+        return unless valid?
+
+        run_tests
+      end
+
+      def parse_value(value)
+        @value = value
+
+        @test_repos, @repos = value.split(/\s+including\s+/)
+                                   .map { |repo_list| repo_list.split(",") }
+
+        # Add the identifier for the PR for this comment to @repos here
+        @repos ||= []
+        @repos  << "#{issue.repo_name}##{issue.number}"
+
+        @test_repos.map! { |repo_name| normalize_repo_name(repo_name) }
+        @repos.map!      { |repo_name| normalize_repo_name(repo_name) }
+      end
+
+      def normalize_repo_name(repo)
+        repo.include?("/") ? repo : "#{issue.organization_name}/#{repo}"
+      end
+
+      def valid?
+        unless issue.pull_request?
+          issue.add_comment("@#{issuer} 'run-test(s)' command is only valid on pull requests, ignoring...")
+          return false
+        end
+
+        unless repos_valid?(@test_repos + @repos)
+          issue.add_comment("@#{issuer} '#{value}' is an invalid repo, ignoring...")
+
+          return false
+        end
+
+        true
+      end
+
+      REPO_ONLY_REGEXP = /(?:[^\/]+\/)?(?<REPO_NAME_ONLY>[^@#]+)/.freeze
+      def repos_valid?(repos)
+        repos.all? do |repo|
+          name_only = repo.match(REPO_ONLY_REGEXP)[:REPO_NAME_ONLY]
+          GithubService.org_repos(issue.organization_name).include?(name_only)
+        end
+      end
+
+      ##### run_tests steps #####
+
+      # Clone repo (if needed) and initialize @rugged_repo
+      def clone_repo_if_necessary
+        repo_path = self.class.test_repo_clone_dir
+        if Dir.exist?(self.class.test_repo_clone_dir)
+          @rugged_repo = Rugged::Repository.new(repo_path)
+        else
+          url = self.class.test_repo_url
+          @rugged_repo = Rugged::Repository.clone_at(url, repo_path)
+        end
+        git_fetch
+      end
+
+      def create_cross_repo_test_branch
+        rugged_repo.create_branch(branch_name, "origin/master")
+        rugged_repo.checkout(branch_name)
+      end
+
+      def update_travis_yaml_content
+        Dir.chdir(@rugged_repo.workdir) do
+          content = YAML.load_file(".travis.yml")
+
+          content["env"] = {} unless content["env"]
+          content["env"]["global"] = ["REPOS=#{repos.join(',')}"]
+          content["env"]["matrix"] = test_repos.map { |repo| "TEST_REPO=#{repo}" }
+
+          File.write('.travis.yml', content.to_yaml)
+        end
+      end
+
+      def commit_travis_yaml_changes
+        index = rugged_repo.index
+        index.add('.travis.yml')
+        index.write
+
+        # rubocop:disable: Rails/TimeZone
```

Comment:
Could you use `Time.now.utc` instead of disabling the rubocop check?

Reply:
Yeah, probably.  I was a little weirded out doing that with a `git commit` versus something in the DB, but looking back, using `.utc` probably shouldn't be a problem.